batavia/types/Str.js

```diff
@@ -233,7 +233,7 @@ String.prototype.__iadd__ = function(other) {
 };
 
 String.prototype.__isub__ = function(other) {
-    throw new batavia.builtins.NotImplementedError("String.__isub__ has not been implemented");
+    throw new batavia.builtins.TypeError("unsupported operand type(s) for -=: 'str' and '" + batavia.type_name(other) +"'")
```

Comment:
Won't the behavior be the same for both branches of the if statement? If `other` is null or NoneType, then `batavia.type_name(other)` will return `"NoneType"`, and the result would be the same as if lines 225 and 230 were duplicates.

Why would we want to split it up into branches when all the branches would do the same thing?